File: 0134-gas-station/0134-gas-station.go
func canCompleteCircuit(gas []int, cost []int) int {
    tank, ind, tot_cost, tot_gas := 0,0,0,0
    for i := range(gas){
        tot_gas += gas[i]
        tot_cost += cost[i]
        tank += gas[i] - cost[i]
        if tank < 0{
            tank = 0
            ind = i+1
        }
    }
    if tot_gas < tot_cost{
        return -1
    }else{
        return ind
    }


}

File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        if sum(gas) < sum(cost):
            return -1
        start = 0 
        curr = 0 
        for i in range(len(gas)):
            net = gas[i] - cost[i]
            curr += net
            if curr < 0:
                start = i +1
                curr = 0
        return start 